src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor_state.go

```diff
@@ -98,36 +99,104 @@ func (st *StoragePowerActorState_I) _selectMinersToSurprise(challengeCount int,
-	activeSectorWeight block.SectorWeight, inactiveSectorWeight block.SectorWeight, ok bool) {
+func (st *StoragePowerActorState_I) _getPowerTotalForMiner(minerAddr addr.Address) (
+	power block.StoragePower, ok bool) {
 
-	powerEntry, found := st.PowerTable()[minerAddr]
+	minerPower, found := st.PowerTable()[minerAddr]
 	if !found {
-		return block.SectorWeight(0), block.SectorWeight(0), found
+		return block.StoragePower(0), found
 	}
 
-	return powerEntry.ActiveSectorWeight(), powerEntry.InactiveSectorWeight(), true
-
+	return minerPower, true
 }
 
-func (st *StoragePowerActorState_I) GetCurrPledgeForMiner(minerAddr addr.Address) (currPledge actor.TokenAmount, ok bool) {
+func (st *StoragePowerActorState_I) _getCurrPledgeForMiner(minerAddr addr.Address) (currPledge actor.TokenAmount, ok bool) {
 	return actor_util.BalanceTable_GetEntry(st.EscrowTable(), minerAddr)
 }
 
-func (st *StoragePowerActorState_I) _getStorageFaultSlashPledgePercent(faultType sector.StorageFaultType) int {
-	PARAM_FINISH() // TODO: instantiate these placeholders
-	panic("")
+func (st *StoragePowerActorState_I) _addClaimedPowerForSector(minerAddr addr.Address, storageWeightDesc SectorStorageWeightDesc) {
+	// Note: The following computation does not use any of the dynamic information from CurrIndices();
+	// it depends only on storageWeightDesc. This means that the power of a given storageWeightDesc
+	// does not vary over time, so we can avoid continually updating it for each sector every epoch.
+	//
+	// The function is located in the indices module temporarily, until we find a better place for
+	// global parameterization functions.
+	sectorPower := indices.ConsensusPowerForStorageWeight(storageWeightDesc)
+
+	currentPower, ok := st.ClaimedPower()[minerAddr]
+	Assert(ok)
+	st._setClaimedPowerEntryInternal(minerAddr, currentPower+sectorPower)
+	st._updatePowerEntriesFromClaimedPower(minerAddr)
+}
+
+func (st *StoragePowerActorState_I) _deductClaimedPowerForSectorAssert(minerAddr addr.Address, storageWeightDesc SectorStorageWeightDesc) {
+	// Note: The following computation does not use any of the dynamic information from CurrIndices();
+	// it depends only on storageWeightDesc. This means that the power of a given storageWeightDesc
+	// does not vary over time, so we can avoid continually updating it for each sector every epoch.
+	//
+	// The function is located in the indices module temporarily, until we find a better place for
+	// global parameterization functions.
+	sectorPower := indices.ConsensusPowerForStorageWeight(storageWeightDesc)
+
+	currentPower, ok := st.ClaimedPower()[minerAddr]
+	Assert(ok)
+	st._setClaimedPowerEntryInternal(minerAddr, currentPower-sectorPower)
+	st._updatePowerEntriesFromClaimedPower(minerAddr)
+}
+
+func (st *StoragePowerActorState_I) _updatePowerEntriesFromClaimedPower(minerAddr addr.Address) {
+	claimedPower, ok := st.ClaimedPower()[minerAddr]
+	Assert(ok)
+
+	nominalPower := claimedPower
+	if st.PoStDetectedFaultMiners()[minerAddr] {
```

Comment:
will good to include a comment here on what nominal power is and what it is used for